test/helpers/restricted-nori.ts

```diff
@@ -0,0 +1,140 @@
+    tokenIds = await Promise.all(
+      removalDataToList.map((removalData, index) => {
+        return createRemovalTokenId({
+          removal,
+          hre,
+          removalData: {
+            supplierAddress: removalData.supplier ?? supplier,
+            vintage: removalData.vintage ?? defaultStartingVintage + index,
+          },
+        });
+      })
+    );
+    const removalBalances = removalDataToList.map((removalData) =>
+      hre.ethers.utils.parseUnits(removalData.amount.toString())
+    );
+
+    const packedData = hre.ethers.utils.defaultAbiCoder.encode(
+      ['uint256', 'uint256', 'address', 'bool'],
+      [projectId, scheduleStartTime, fifoMarket.address, true]
+    );
+    expect(
+      await removal.mintBatch(supplier, removalBalances, tokenIds, packedData)
+    ).to.emit(rNori, 'ScheduleCreated');
+  }
+
+  return {
+    listedRemovalIds: tokenIds,
+    projectId,
+    scheduleStartTime,
+  };
+};
+
+export const formatTokensReceivedUserData = (removalId: BigNumber): any => {
+  return hre.ethers.utils.defaultAbiCoder.encode(['uint256'], [removalId]);
+};
+
+export const restrictRemovalProceeds = async ({
+  testSetup,
+  removalIds,
+  removalAmountsToRestrict,
+}: {
+  testSetup: Awaited<ReturnType<typeof setupTest>>;
+  removalIds: BigNumber[];
+  removalAmountsToRestrict: number[];
+}): Promise<void> => {
+  const { rNori, bpNori } = testSetup;
+  await Promise.all(
+    removalIds.map((id, index) => {
+      const userData = formatTokensReceivedUserData(id);
+      return bpNori.send(
+        rNori.address,
+        removalAmountsToRestrict[index],
+        userData
+      );
+    })
+  );
+};
+
+export const compareScheduleDetailForAddressStructs = (
+  receivedScheduleDetail: RestrictedNORI.ScheduleDetailForAddressStruct,
+  expectedScheduleDetail: Partial<RestrictedNORI.ScheduleDetailForAddressStruct>
+): void => {
+  const keys = [
+    'tokenHolder',
+    'scheduleTokenId',
+    'startTime',
+    'endTime',
+    'balance',
+    'claimableAmount',
+    'claimedAmount',
+    'quantityRevoked',
+    'exists',
+  ] as const;
+  for (const key of keys) {
+    if (key in expectedScheduleDetail) {
+      expect(receivedScheduleDetail[key]).to.equal(expectedScheduleDetail[key]);
+    }
+  }
+};
+
+export const compareScheduleSummaryStructs = (
+  receivedScheduleSummary: RestrictedNORI.ScheduleSummaryStruct,
+  expectedScheduleSummary: Partial<RestrictedNORI.ScheduleSummaryStruct>
+): void => {
+  const keys = [
+    'scheduleTokenId',
+    'startTime',
+    'endTime',
+    'totalSupply',
+    'totalClaimableAmount',
+    'totalQuantityRevoked',
+    'exists',
+  ] as const;
+  for (const key of keys) {
+    if (key in expectedScheduleSummary) {
+      expect(receivedScheduleSummary[key]).to.equal(
+        expectedScheduleSummary[key]
+      );
+    }
+  }
+  if (expectedScheduleSummary.tokenHolders !== undefined) {
+    expect(receivedScheduleSummary.tokenHolders).to.have.members(
```

Comment:
don't love this as it breaks [no-conditional-expect](https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-conditional-expect.md)